hazelcast/src/main/java/com/hazelcast/cache/impl/CachePartitionSegment.java

```diff
@@ -91,11 +104,28 @@ public void destroy() {
         clear();
     }
 
-    public boolean hasAnyRecordStore() {
-        return !recordStores.isEmpty();
+    public void close() {
+        synchronized (mutex) {
+            for (ICacheRecordStore store : recordStores.values()) {
+                store.close();
+            }
+        }
+        recordStores.clear();
     }
 
-    public boolean hasRecordStore(String name) {
-        return recordStores.containsKey(name);
+    protected static void transferRecordData(ICacheRecordStore src, ICacheRecordStore dest) {
```

Comment:
:+1: BTW `expiryPolicyToTTL()` and `expiryPolicyToTTL()` methods are used only for enterprise side. We can move both of them to enterprise also